protobuf/key_attributes.proto

```diff
@@ -18,62 +18,59 @@ syntax = "proto3";
 
 package key_attributes;
 
+message KeyAttributesProto {
+  KeyType key_type = 1;
+  oneof algorithm {
```

Comment:
Maybe would be nice to have them in the same order than in the spec. It makes it easier to compare if both definitions are the same.

Reply:
What about the wildcard one? `PSA_ALG_ANY_HASH` is given as example.

Reply:
Whereas the name is `algorithm` it seems like this field encode the *permitted* algorithms that this key can use. Would be nice to either add a comment to make sure it is understood, write that in the docs, or even rename this field `permitted_key_algorithm`

Reply:
Oh boy, this would bring on a whole bunch of changes...
* we should probably support a "wildcard" algorithm, but I think for us it makes more sense for it to be a variant of the `Hash` enum - `ANY_HASH`. Mbed Crypto only accepts hash as a wild card, either just for hashing or for sign&hash. If in the future other categories will get a wildcard, we can add it to their enum
* this means that we now have to add a hash param to the `SignHash` and `SignMessage` operations and make sure the requested algo is allowed.
* putting is as a not here though not directly related - we need to change the `Sign` operation to `SignHash`
* if there was only one permitted algorithm per key, then calling it `algorithm` actually makes sense, but with the "wildcard" `permitted_algorithm` is better